### PR TITLE
Consistently reset filters

### PR DESCRIPTION
**Pull request by damb**
**Features and Changes**:
- Consistently reset the filter state when (re)configuring the sampling frequency. (Otherwise, calling `Seiscomp::Processing::WaveformProcessor::initFilter()` in `Seiscomp::Processing::WaveformProcessor::setupStream()` may leave filters with state in case of changing sampling frequencies.)
- Consistently apply SeisComP coding conventions with regards to virtual member functions and overrides.

Please review carefully.

### CODE REVIEW

File: libs/seiscomp/math/filter/iirdifferentiate.cpp [hunk 1]
```diff
@@ -30,15 +30,13 @@ namespace Filtering
...
-	reset();
 	setSamplingFrequency(fsamp);
 }
 
 
 template <typename T>
 IIRDifferentiate<T>::IIRDifferentiate(const IIRDifferentiate<T> &other) {
-	_fsamp = 0.0;
-	reset();
+  setSamplingFrequency(other._fsamp);
```
Comment (gempa-jabe) [inline libs/seiscomp/math/filter/iirdifferentiate.cpp:39]: If the compiler initializes _fsamp with 0.0 then this function does nothing and the other reset states are still unconfigured. The original two lines stable in that regard.

File: libs/seiscomp/math/filter/iirintegrate.cpp [hunk 2]
```diff
@@ -71,9 +71,13 @@ void IIRIntegrate<T>::setSamplingFrequency(double fsamp) {
 
 	double dt = 1./fsamp;
 
-	_a0 = _ia0 * dt;
+  double a0 = _ia0 * dt;
```
Comment (gempa-jabe) [inline libs/seiscomp/math/filter/iirintegrate.cpp:74]: Does this block intent to check if the sampling frequency has changed? Why not just checking it at the top of the function as with all other filters?

File: libs/seiscomp/math/filter/taper.cpp [hunk 3]
```diff
@@ -34,6 +34,17 @@ InitialTaper<TYPE>::InitialTaper(double taperLength, TYPE offset, double fsamp)
 		setSamplingFrequency(fsamp);
 }
 
+template<typename TYPE>
+void InitialTaper<TYPE>::setSamplingFrequency(double fsamp) {
+	if (_samplingFrequency == fsamp) return;
```
Comment (gempa-jabe) [inline libs/seiscomp/math/filter/taper.cpp:39]: Please add spaces between parenthesis: `if ( a )`

**Review by gempa-jabe — CHANGES_REQUESTED** [review]

**Comment by gempa-jabe** [comment]
Thanks a lot for the work on this. Cleaning up the filter code has been on the list for some time.

File: libs/seiscomp/math/filter/iirintegrate.cpp (hunk 2 above)
Comment (damb) [reply libs/seiscomp/math/filter/iirintegrate.cpp:74]: Yes. I didn't like it, neither. Instead, it is much cleaner to store the sampling frequency as a member variable and do the check as with all other filters.

Comment (gempa-jabe) [reply libs/seiscomp/math/filter/iirintegrate.cpp]: OK, I see. Adding _fsamp will break binary compability. So this might be the best approach.

Comment (damb) [reply libs/seiscomp/math/filter/iirintegrate.cpp]: I went for the original attempt with a minor change.

>  Adding _fsamp will break binary compability.

Again, I learned something :) ...

**Comment by gempa-jabe** [comment]
I have checked the code now multiple times and it should not break anything. I am going to merge it. Thanks again.

**Pull request merged**